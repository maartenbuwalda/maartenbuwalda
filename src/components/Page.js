import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'
import colors from '../constants/colors'

const Article = styled.article`
  width: 100%;
  max-width: 30rem;
  padding: 1rem;
  margin: 1rem;
  background-color: ${colors.white};
`

const Page = ({ children }) => {
  return (
    <Article>
      {children}
    </Article>
  )
}

Page.propTypes = {
  children: PropTypes.node,
}

export default Page
