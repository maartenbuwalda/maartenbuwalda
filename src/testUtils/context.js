const context = {
  theme: {},
  updateTheme: jest.fn(),
}

export default context
