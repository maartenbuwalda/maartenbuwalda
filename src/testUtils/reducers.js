export const reducerReturnsInitialState = (reducer, initialState) => {
  it('should succeed: return the initialState', () => {
    expect(reducer(undefined, {})).toEqual(initialState)
  })
}
