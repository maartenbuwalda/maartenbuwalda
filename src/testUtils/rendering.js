import toJson from 'enzyme-to-json'
import { shallow, mount } from 'enzyme'
import 'jest-styled-components'

export const componentRenders = component => {
  it('renders correctly', () => {
    const shallowed = shallow(component)
    expect(toJson(shallowed)).toMatchSnapshot()
  })
}

export const componentMounts = component => {
  it('mounts correctly', () => {
    const mounted = mount(component)
    expect(toJson(mounted)).toMatchSnapshot()
  })
}
