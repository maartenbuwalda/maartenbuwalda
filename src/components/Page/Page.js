import React from 'react'
import styled from 'styled-components'
import HeaderImage from '../../components/HeaderImage'
import PropTypes from 'prop-types'
import { ColorContext } from '../../context/color'
import colors from '../../constants/colors'
import { sizes } from '../../constants/mediaQueries'

const PageContent = styled.article`

  section {
    padding: 4rem 2rem;
    transition: background-color 1s;
  }

  @media(min-width: ${sizes.m}) {
    section {
      padding: 4rem;
    }
  }

  > section:nth-child(even), .colored {
    background-color: rgba(${({ theme }) => theme.rgb.colorMutedLight}, 0.2);
  }

  h2 {
    font-size: 2rem;
    margin: 0;
  }
  
  h3 {
    font-size: 1.5rem;

    &:first-of-type {
      margin-top: 0;
    }
  }

  ul {
    list-style-type: disc;
    padding-left: 1rem;

    li {
      margin-bottom: 1rem;
      font-size: 1.2rem;
    }
  }

  p {
    line-height: 1.7;
    font-size: 1.2rem;
    color: ${colors.fadedBlack}
  }
`

const Page = ({ page }) => {
  return (
    <ColorContext.Consumer>
      {({ updateTheme, theme }) => (
        <>
          {page.image && (
            <HeaderImage
              updateTheme={updateTheme}
              image={page.image}
              title={page.title}
            />
          )}
          {page.content && (
            <PageContent theme={theme}>
              {page.content}
            </PageContent>
          )}
        </>
      )}
    </ColorContext.Consumer>
  )
}

Page.propTypes = {
  page: PropTypes.shape({
    image: PropTypes.string,
    title: PropTypes.string,
    content: PropTypes.node,
  })
}

export default Page
