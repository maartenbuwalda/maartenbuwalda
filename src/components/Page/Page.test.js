import React from 'react'
import Page from './index'
import { componentMounts } from '../../testUtils/rendering'

describe('Page component', () => {
  const component = <Page page={{
    image: 'test',
    content: 'test',
  }} />
  componentMounts(component)
})
