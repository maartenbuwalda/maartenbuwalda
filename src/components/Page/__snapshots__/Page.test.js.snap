// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page component mounts correctly 1`] = `
.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 1rem;
  width: 3rem;
  height: 3rem;
}

.c3 svg {
  width: 100%;
  height: 100%;
  fill: rgba(255,255,255,0.8);
  -webkit-transition: fill .3s;
  transition: fill .3s;
}

.c3 svg path {
  -webkit-transition: all .3s;
  transition: all .3s;
  opacity: 1;
  -webkit-transform: scaleY(0.9);
  -ms-transform: scaleY(0.9);
  transform: scaleY(0.9);
}

.c3 svg path[data-appear=true] {
  -webkit-transform: scaleY(1);
  -ms-transform: scaleY(1);
  transform: scaleY(1);
  opacity: 1;
}

.c2 {
  display: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: absolute;
  -webkit-transition: background-color 1s;
  transition: background-color 1s;
  top: 1rem;
  left: 1rem;
}

.c2 h1 {
  color: #ffffff;
}

.c4 {
  -webkit-transition: all .3s;
  transition: all .3s;
  z-index: 2;
  padding: .5rem 1rem;
  margin-bottom: 1rem;
}

.c4 h1 {
  -webkit-transition: color .3s;
  transition: color .3s;
  font-size: 2rem;
  font-weight: normal;
  margin: 0;
  color: #ffffff;
}

.c0 {
  min-height: 15rem;
  width: 100%;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c1 {
  position: absolute;
  background-image: url(test);
  background-repeat: no-repeat;
  background-position: center center;
  height: 100%;
  width: 100%;
  background-size: cover;
}

.c5 {
  text-align: center;
}

.c5 section {
  padding: 2rem;
  -webkit-transition: background-color 1s;
  transition: background-color 1s;
}

.c5 > section:nth-child(even) {
  background-color: rgba(,0.1);
}

.c5 h2 {
  font-size: 2rem;
  margin: 0;
}

.c5 p {
  line-height: 1.7;
  font-size: 1.2rem;
  color: rgba(0,0,0,0.8);
}

@media (min-width:768px) {
  .c4 h1 {
    font-size: 3rem;
  }
}

@media (min-width:768px) {
  .c0 {
    min-height: 30rem;
  }
}

@media (min-width:768px) {
  .c5 section {
    padding: 4rem;
  }
}

<Page
  page={
    Object {
      "content": "test",
      "image": "test",
    }
  }
>
  <HeaderImage
    image="test"
    updateTheme={[Function]}
  >
    <styled.div>
      <StyledComponent
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bxivhb",
              "isStatic": true,
              "lastClassName": "c0",
              "rules": Array [
                "
  min-height: 15rem;
  width: 100%;
  position: relative;
  display: flex;
  justify-content: center;
  align-items: flex-end;

  @media (min-width: ",
                "768px",
                ") {
    min-height: 30rem;
    /* justify-content: flex-end; */
  }
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bxivhb",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
      >
        <div
          className="c0"
        >
          <styled.div
            image="test"
          >
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-ifAKCX",
                    "isStatic": false,
                    "lastClassName": "c1",
                    "rules": Array [
                      "
  position: absolute;
  background-image: url(",
                      [Function],
                      ");
  background-repeat: no-repeat;
  background-position: center center;
  height: 100%;
  width: 100%;
  background-size: cover;
",
                    ],
                  },
                  "displayName": "styled.div",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-ifAKCX",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              image="test"
            >
              <div
                className="c1"
              />
            </StyledComponent>
          </styled.div>
          <styled.div
            theme={
              Object {
                "hex": Object {
                  "colorDark": "",
                  "colorLight": "",
                  "colorMutedDark": "",
                  "colorMutedLight": "",
                  "colorMutedRegular": "",
                  "colorRegular": "",
                },
                "rgb": Object {
                  "colorDark": "",
                  "colorLight": "",
                  "colorMutedDark": "",
                  "colorMutedLight": "",
                  "colorMutedRegular": "",
                  "colorRegular": "",
                },
              }
            }
          >
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bwzfXH",
                    "isStatic": false,
                    "lastClassName": "c2",
                    "rules": Array [
                      "
  display: none;
  align-items: center;
  position: absolute;
  transition: background-color 1s;
  top: 1rem;
  left: 1rem;
  background-color: ",
                      [Function],
                      ";

  h1 {
    color: ",
                      "#ffffff",
                      "
  }
",
                    ],
                  },
                  "displayName": "styled.div",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bwzfXH",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              theme={
                Object {
                  "hex": Object {
                    "colorDark": "",
                    "colorLight": "",
                    "colorMutedDark": "",
                    "colorMutedLight": "",
                    "colorMutedRegular": "",
                    "colorRegular": "",
                  },
                  "rgb": Object {
                    "colorDark": "",
                    "colorLight": "",
                    "colorMutedDark": "",
                    "colorMutedLight": "",
                    "colorMutedRegular": "",
                    "colorRegular": "",
                  },
                }
              }
            >
              <div
                className="c2"
              >
                <Logo
                  size="3rem"
                  white={true}
                >
                  <styled.div
                    size="3rem"
                    theme={
                      Object {
                        "hex": Object {
                          "colorDark": "",
                          "colorLight": "",
                          "colorMutedDark": "",
                          "colorMutedLight": "",
                          "colorMutedRegular": "",
                          "colorRegular": "",
                        },
                        "rgb": Object {
                          "colorDark": "",
                          "colorLight": "",
                          "colorMutedDark": "",
                          "colorMutedLight": "",
                          "colorMutedRegular": "",
                          "colorRegular": "",
                        },
                      }
                    }
                    white={true}
                  >
                    <StyledComponent
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "sc-bdVaJa",
                            "isStatic": false,
                            "lastClassName": "c3",
                            "rules": Array [
                              "
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 1rem;
  width: ",
                              [Function],
                              ";
  height: ",
                              [Function],
                              ";

  svg {
    width: 100%;
    height: 100%;
    fill: ",
                              [Function],
                              ";
    transition: fill .3s;

    path {
      transition: all .3s;
      opacity: ",
                              [Function],
                              ";
      transform: scaleY(0.9);

      &[data-appear=true] {
        transform: scaleY(1);
        opacity: 1;
      }
    }
  }
",
                            ],
                          },
                          "displayName": "styled.div",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "sc-bdVaJa",
                          "target": "div",
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                      size="3rem"
                      theme={
                        Object {
                          "hex": Object {
                            "colorDark": "",
                            "colorLight": "",
                            "colorMutedDark": "",
                            "colorMutedLight": "",
                            "colorMutedRegular": "",
                            "colorRegular": "",
                          },
                          "rgb": Object {
                            "colorDark": "",
                            "colorLight": "",
                            "colorMutedDark": "",
                            "colorMutedLight": "",
                            "colorMutedRegular": "",
                            "colorRegular": "",
                          },
                        }
                      }
                      white={true}
                    >
                      <div
                        className="c3"
                        size="3rem"
                      >
                        <svg
                          preserveAspectRatio="xMidYMid meet"
                          version="1.0"
                          viewBox="0 0 509.000000 509.000000"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <g
                            stroke="none"
                            transform="translate(0.000000,509.000000) scale(0.100000,-0.100000)"
                          >
                            <path
                              d="M2065 4954 c-269 -73 -507 -138 -527 -144 l-38 -11 0 -554 0 -555 150 0 150 0 0 435 c0 239 2 435 4 435 2 0 137 36 299 80 162 44 298 80 301 80 3 0 6 -49 6 -110 l0 -110 150 0 150 0 0 110 c0 61 3 110 8 110 7 0 381 -104 505 -141 l67 -20 0 -434 0 -435 150 0 150 0 0 554 c0 520 -1 555 -17 560 -150 45 -1005 286 -1010 285 -5 -1 -228 -61 -498 -135z"
                            />
                            <path
                              d="M700 3585 l0 -805 150 0 150 0 0 655 0 655 195 0 195 0 0 150 0 150 -345 0 -345 0 0 -805z"
                            />
                            <path
                              d="M1910 4240 l0 -150 635 0 635 0 0 150 0 150 -635 0 -635 0 0 -150z"
                            />
                            <path
                              d="M3700 4240 l0 -150 200 0 200 0 0 -655 0 -655 150 0 150 0 0 805 0 805 -350 0 -350 0 0 -150z"
                            />
                            <path
                              d="M2410 3835 l0 -145 150 0 150 0 0 145 0 145 -150 0 -150 0 0 -145z"
                            />
                            <path
                              d="M287 3573 c-2 -5 -68 -242 -147 -527 l-142 -519 144 -516 145 -516 151 -3 152 -3 0 150 0 150 -32 3 -32 3 -78 280 c-43 154 -78 283 -78 288 0 4 230 7 510 7 l510 0 0 150 0 150 -514 2 -513 3 82 300 82 300 32 3 31 3 0 150 0 149 -149 0 c-83 0 -152 -3 -154 -7z"
                            />
                            <path
                              d="M1110 3430 l0 -150 140 0 140 0 0 150 0 150 -140 0 -140 0 0 -150z"
                            />
                            <path
                              d="M1500 2740 l0 -840 150 0 150 0 0 690 0 690 305 0 305 0 0 -690 0 -690 150 0 150 0 0 685 0 685 290 0 290 0 0 -685 0 -685 150 0 150 0 0 840 0 840 -1045 0 -1045 0 0 -840z"
                            />
                            <path
                              d="M3700 3425 l0 -155 145 0 145 0 0 155 0 155 -145 0 -145 0 0 -155z"
                            />
                            <path
                              d="M4510 3430 l0 -150 24 0 c25 0 14 34 184 -592 4 -17 -25 -18 -507 -18 l-511 0 0 -150 0 -150 510 0 c281 0 510 -3 510 -8 0 -4 -36 -133 -80 -287 -44 -154 -80 -281 -80 -282 0 -2 -11 -3 -25 -3 l-25 0 0 -150 0 -150 145 0 144 0 10 38 c6 20 71 250 145 510 l134 473 -139 517 c-77 284 -142 525 -145 535 -5 15 -21 17 -150 17 l-144 0 0 -150z"
                            />
                            <path
                              d="M700 1470 l0 -790 345 0 345 0 0 155 0 155 -195 0 -195 0 0 635 0 635 -150 0 -150 0 0 -790z"
                            />
                            <path
                              d="M4100 1625 l0 -635 -200 0 -200 0 0 -150 0 -150 350 0 350 0 0 785 0 785 -150 0 -150 0 0 -635z"
                            />
                            <path
                              d="M1110 1640 l0 -150 268 0 c147 0 439 -3 650 -7 l382 -6 0 -189 0 -188 150 0 150 0 0 188 0 189 362 6 c199 4 487 7 640 7 l278 0 0 150 0 150 -1440 0 -1440 0 0 -150z"
                            />
                            <path
                              d="M1502 833 l3 -546 524 -144 524 -144 516 144 516 145 3 546 2 546 -150 0 -150 0 0 -425 0 -424 -67 -20 c-108 -32 -501 -141 -507 -141 -4 0 -6 47 -6 105 l0 105 -150 0 -149 0 -3 -108 -3 -109 -265 73 c-146 40 -282 78 -302 84 l-38 11 0 424 0 425 -150 0 -150 0 2 -547z"
                            />
                            <path
                              d="M1910 835 l0 -155 635 0 635 0 0 155 0 155 -635 0 -635 0 0 -155z"
                            />
                          </g>
                        </svg>
                      </div>
                    </StyledComponent>
                  </styled.div>
                </Logo>
              </div>
            </StyledComponent>
          </styled.div>
          <styled.div
            theme={
              Object {
                "hex": Object {
                  "colorDark": "",
                  "colorLight": "",
                  "colorMutedDark": "",
                  "colorMutedLight": "",
                  "colorMutedRegular": "",
                  "colorRegular": "",
                },
                "rgb": Object {
                  "colorDark": "",
                  "colorLight": "",
                  "colorMutedDark": "",
                  "colorMutedLight": "",
                  "colorMutedRegular": "",
                  "colorRegular": "",
                },
              }
            }
          >
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-htpNat",
                    "isStatic": false,
                    "lastClassName": "c4",
                    "rules": Array [
                      "
  transition: all .3s;
  z-index: 2;
  padding: .5rem 1rem;
  background-color: ",
                      [Function],
                      ";
  margin-bottom: 1rem;

  h1 {
    transition: color .3s;
    font-size: 2rem;
    font-weight: normal;
    margin: 0;
    color: ",
                      "#ffffff",
                      ";

    @media (min-width: ",
                      "768px",
                      ") {
      font-size: 3rem;
    }
  }
",
                    ],
                  },
                  "displayName": "styled.div",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-htpNat",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              theme={
                Object {
                  "hex": Object {
                    "colorDark": "",
                    "colorLight": "",
                    "colorMutedDark": "",
                    "colorMutedLight": "",
                    "colorMutedRegular": "",
                    "colorRegular": "",
                  },
                  "rgb": Object {
                    "colorDark": "",
                    "colorLight": "",
                    "colorMutedDark": "",
                    "colorMutedLight": "",
                    "colorMutedRegular": "",
                    "colorRegular": "",
                  },
                }
              }
            >
              <div
                className="c4"
              >
                <h1 />
              </div>
            </StyledComponent>
          </styled.div>
        </div>
      </StyledComponent>
    </styled.div>
  </HeaderImage>
  <styled.div
    theme={
      Object {
        "hex": Object {
          "colorDark": "",
          "colorLight": "",
          "colorMutedDark": "",
          "colorMutedLight": "",
          "colorMutedRegular": "",
          "colorRegular": "",
        },
        "rgb": Object {
          "colorDark": "",
          "colorLight": "",
          "colorMutedDark": "",
          "colorMutedLight": "",
          "colorMutedRegular": "",
          "colorRegular": "",
        },
      }
    }
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-EHOje",
            "isStatic": false,
            "lastClassName": "c5",
            "rules": Array [
              "
  text-align: center;

  section {
    padding: 2rem;
    transition: background-color 1s;
  }

  @media(min-width: ",
              "768px",
              ") {
    section {
      padding: 4rem;
    }
  }

  > section:nth-child(even) {
    background-color: rgba(",
              [Function],
              ", 0.1);
  }

  h2 {
    font-size: 2rem;
    margin: 0;
  }

  p {
    line-height: 1.7;
    font-size: 1.2rem;
    color: ",
              "rgba(0, 0, 0, 0.8)",
              "
  }
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-EHOje",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      theme={
        Object {
          "hex": Object {
            "colorDark": "",
            "colorLight": "",
            "colorMutedDark": "",
            "colorMutedLight": "",
            "colorMutedRegular": "",
            "colorRegular": "",
          },
          "rgb": Object {
            "colorDark": "",
            "colorLight": "",
            "colorMutedDark": "",
            "colorMutedLight": "",
            "colorMutedRegular": "",
            "colorRegular": "",
          },
        }
      }
    >
      <div
        className="c5"
      >
        test
      </div>
    </StyledComponent>
  </styled.div>
</Page>
`;
